src/components/modals/CancelWithdrawalModal.tsx

```diff
@@ -0,0 +1,140 @@
+        // onClose();
+      } catch (e: any) {
+        showErrorToast(`${e}`);
+      } finally {
+        setShowBlockingMessageModal(false);
+      }
+    }
+  };
+
+  return (
+    <>
+      <BaseModal onClose={onClose} useDefaultPadding={false}>
+        <div className="w-[28.4375rem] text-left">
+          <div className="px-8  pb-4 pt-6">
+            <div className="text-lg text-high">Cancel Pending Withdrawal</div>
+          </div>
+
+          <div className="border-y border-grey-800 p-8 text-sm text-mid">
+            <div>
+              This action will cancel your withdrawal and return its stake to
+              the original gateway. This action cannot be undone.
+            </div>
+          </div>
+
+          <div className="px-8 pb-8 pt-6">
+            <div className="mb-6 flex flex-col items-center gap-2 text-sm text-mid">
+              <div>
+                Please type &quot;CONFIRM&quot; in the text box to proceed.
+              </div>
+              <input
+                type="text"
+                onChange={(e) => setConfirmText(e.target.value)}
+                className={
+                  'h-7 w-full rounded-md border border-grey-700 bg-grey-1000 p-3 text-sm text-mid outline-none placeholder:text-grey-400 focus:text-high'
+                }
+                value={confirmText}
+              />
+            </div>
+
+            <div className="flex grow justify-center">
+              <Button
+                onClick={processCancelWithdrawal}
+                buttonType={ButtonType.PRIMARY}
+                title="Leave Network"
+                text={<div className="py-2">Cancel Withdrawal</div>}
+                className={`w-full ${!termsAccepted && 'pointer-events-none opacity-30'}`}
+              />
+            </div>
+          </div>
+        </div>
+      </BaseModal>
+      {showBlockingMessageModal && (
+        <BlockingMessageModal
+          onClose={() => setShowBlockingMessageModal(false)}
+          message="Sign the following data with your wallet to proceed."
+        ></BlockingMessageModal>
+      )}
+      {showSuccessModal && (
+        <SuccessModal
+          onClose={() => {
+            setShowSuccessModal(false);
+            onClose();
+          }}
+          title="Confirmed"
+          bodyText={
+            <div className="mb-8 text-sm text-mid">
+              <div>You have successfully canceled the withdrawal.</div>
+              <div className="my-2 flex flex-col justify-center gap-2">
+                <div>Transaction ID:</div>
+                <button
+                  className="flex items-center justify-center"
+                  title="View transaction on ao.link"
+                  onClick={async () => {
+                    window.open(`https://ao.link/#/message/${txid}`, '_blank');
+                  }}
+                >
+                  {txid}
+                  <LinkArrowIcon className="ml-1 size-3" />
+                </button>
```

Comment:
This should probably be a link, not a button

Reply:
There's a problem using links that it doesn't open the link. Leaving here for later reference:

```
<a 
                  href={`https://ao.link/#/message/${txid}`}
                  className="flex items-center justify-center"
                  title="View transaction on ao.link"
                  target='_blank'
                  rel='noreferrer'
                >
                  {txid}
                  <LinkArrowIcon className="ml-1 size-3" />
                </a>
```

Will leave as a button for now as it works and will add a // FIXME comment to investigate later

Reply:
What about the Link component from react-dom?